src/AttachVS/AttachVs.cs

```diff
@@ -0,0 +1,309 @@
+                        }
+                        catch
+                        {
+                            return null;
+                        }
+                    }).Where(p => p != null && !p.HasExited).OrderBy(p => p.StartTime).ToList();
+
+                    var firstVs = processes.FirstOrDefault();
+                    Trace($"Found VS {firstVs.Process.Id}");
+                    AttachTo(process, firstVs.Process);
+                }
+                return true;
+            }
+            catch (Exception ex)
+            {
+                Trace($"ERROR: {ex}, {ex.StackTrace}");
+                return false;
+            }
+        }
+
+        private static void AttachTo(Process process, Process vs)
+        {
+            var attached = AttachVs(vs, process.Id);
+            if (attached)
+            {
+                // You won't see this in DebugView++ because at this point VS is already attached and all the output goes into Debug window in VS.
+                Trace($"SUCCESS: Attached process: {process.ProcessName} ({process.Id})");
+            }
+            else
+            {
+                Trace($"FAIL: Could not attach process: {process.ProcessName} ({process.Id})");
+            }
+        }
+
+        private static bool AttachVs(Process vs, int pid)
+        {
+            IBindCtx bindCtx = null;
+            IRunningObjectTable runninObjectTable = null;
+            IEnumMoniker enumMoniker = null;
+            try
+            {
+                var r = CreateBindCtx(0, out bindCtx);
+                Marshal.ThrowExceptionForHR(r);
+                if (bindCtx == null)
+                {
+                    Trace($"BindCtx is null. Cannot attach VS.");
+                    return false;
+                }
+                bindCtx.GetRunningObjectTable(out runninObjectTable);
+                if (runninObjectTable == null)
+                {
+                    Trace($"RunningObjectTable is null. Cannot attach VS.");
+                    return false;
+                }
+
+                runninObjectTable.EnumRunning(out enumMoniker);
+                if (enumMoniker == null)
+                {
+                    Trace($"EnumMoniker is null. Cannot attach VS.");
+                    return false;
+                }
+
+                var dteSuffix = ":" + vs.Id;
+
+                var moniker = new IMoniker[1];
+                while (enumMoniker.Next(1, moniker, IntPtr.Zero) == 0 && moniker[0] != null)
+                {
+                    string dn;
+
+                    moniker[0].GetDisplayName(bindCtx, null, out dn);
+
+                    if (dn.StartsWith("!VisualStudio.DTE.") && dn.EndsWith(dteSuffix))
+                    {
+                        object dte, dbg, lps;
+                        runninObjectTable.GetObject(moniker[0], out dte);
+
+                        for (var i = 0; i < 10; i++)
```

Comment:
It's not clear to me why we do this 10 times ?

Reply:
When the com object is busy doing something else, or blocked by modal window, or attaching other window this will return a com error, that the object is not ready and we need to retry, 10 is arbitrary.